core/src/main/java/org/apache/calcite/rel/rules/JoinToMultiJoinRule.java

```diff
@@ -348,7 +348,7 @@ private void copyOuterJoinInfo(
     if (adjustmentAmount == 0) {
       destJoinSpecs.addAll(srcJoinSpecs);
     } else {
-      int nFields = srcFields.size();
+      int nFields = srcFields == null ? 0 : srcFields.size();
```

Comment:
a little defensive. `assert srcFields != null` would be better, since that is the implicit contract. also `assert destFields != null`